### DISCUSSION TITLE
Tiktok Traffic statistics

### DISCUSSION THREAD

**Discussion by rejrejrej1** [YOURLS/YOURLS#3409] [Q&A]
Hey everyone
start to use YOURLS yesterday
i put my short link in my tiktok profile
and if someone click the url i cant see the click on traffic statistics 
somone know how i can slove this?
thanks

**Comment by logusivam** [comment]
This is a very common issue specifically with **TikTok** (and Instagram, Facebook, etc.) because of how their in-app browser handles link clicks — and the good news is your YOURLS install is very likely working correctly. The clicks just aren't showing up the way you'd expect for a different reason than a typical "no clicks yet" situation.

---

## Why TikTok links specifically cause this

When someone taps a link in your TikTok bio or profile, it does **not** open in their normal browser (Safari, Chrome, etc.). It opens inside **TikTok's own in-app browser** (a WebView embedded in the TikTok app itself). This matters for two reasons:

1. **The click should still register** — YOURLS counts the click and logs the visit as soon as the request hits your server, regardless of which browser/app made the request. If the destination page loaded for the user, the click happened and should be in the database.
2. **In-app browsers behave differently than real browsers** for some tracking signals — particularly referrer headers and sometimes user-agent strings — which can make the entry look incomplete or get miscategorized, even though the click itself was counted.

So the first thing to determine is: **is the click count itself missing, or is it counted but just not showing where you expect?**

---

## Step 1 — Check the basic click counter first

Go to your YOURLS admin dashboard and look at the **main links list** (not the detailed stats page) — specifically the click count number next to your short link.

- **If the number has gone up** (e.g. shows "1" or more) → the click IS being recorded. Your issue is specifically with the **detailed Traffic Statistics page** (graphs, referrer, location), not the click tracking itself. Go to Step 2.
- **If the number is still 0** even after a real click from TikTok → the click isn't being recorded at all. Go to Step 3.

---

## Step 2 — If the click count works but Traffic Statistics page looks empty

This is the more likely scenario. Click on your short link in the admin panel and check the **"Traffic Statistics"** detail page (usually `/admin/stats.php?page=1&...`).

- **"Direct" or no referrer showing** — completely normal. TikTok's in-app browser frequently strips or doesn't send a `Referer` header, especially for links opened from a profile bio. Same with bookmarks or directly typed URLs.
- **Location/country map blank** — if you tested the link yourself, check whether you tested over WiFi at home vs. mobile data. Private/local IPs don't resolve to a country.
- **The graph/chart shows 1 click but details below are sparse** — this is expected; TikTok's WebView doesn't send the same headers a normal mobile browser would, so some fields (referrer, sometimes user agent details) come through empty or generic.

None of this means tracking is broken — it means TikTok's in-app browser sends a stripped-down request compared to a normal browser, so some of YOURLS's stat fields legitimately have nothing to show.

---

## Step 3 — If the click count itself shows 0 (the click isn't registering at all)

If the basic counter isn't moving even after a confirmed click, check these:

**A. Confirm you're testing correctly**
- Click the link from inside the actual TikTok app (not by copying the link out and pasting it into a regular browser) — to reproduce the real scenario.
- Wait a few seconds and refresh your admin panel — there can be a slight delay.

**B. Check if `YOURLS_NOSTATS` or a similar setting is accidentally enabled**

Open `user/config.php` and confirm this line is either absent or set to `false`:

```php
define( 'YOURLS_NOSTATS', false );
```

If it's set to `true`, click tracking is intentionally disabled site-wide.

**C. Check if a plugin is interfering**

If you have any analytics, caching, or security plugins active, temporarily deactivate them one at a time via **Admin → Manage Plugins** and re-test. Some caching/firewall-style plugins can intercept requests before YOURLS logs them.

**D. Confirm the short link itself is correct**

Double check the exact short URL you put in your TikTok bio matches a real, active link in your YOURLS install — a typo or an extra character would mean clicks are hitting a 404 instead of your actual short link, and would never be logged.

---

## Quick way to confirm everything works correctly

Test with a link you control completely:

1. Create a fresh test short link in YOURLS.
2. Put it somewhere you can click it from your phone's TikTok app (e.g. send it to yourself in a TikTok DM, or temporarily put it in your bio).
3. Tap it from inside TikTok on mobile data (not WiFi, to get a real public IP).
4. Check the admin panel immediately after — the basic click counter should increment even if referrer/location stay mostly blank.

If the counter increments, your tracking is working. The "missing" referrer and location detail is just a limitation of what TikTok's in-app browser sends, not a YOURLS problem.

---

## Summary

| Symptom | Likely cause | Action |
|---------|-------------|--------|
| Click counter shows 0 | Tracking genuinely not registering | Check `YOURLS_NOSTATS`, deactivate plugins one by one, confirm link is correct |
| Click counter increments but stats page looks empty | Normal — TikTok in-app browser strips referrer/headers | Nothing to fix; this is expected behavior |
| No location data | Test was from local/WiFi network or in-app browser limits IP exposure | Test from mobile data via the real TikTok app |

Hope this helps — if it answers your question, would you mind clicking **"Mark as answer"** so others searching for the same thing can find it quickly?
